BigQuery/src/Connection/Rest.php

```diff
@@ -65,10 +71,11 @@ public function __construct(array $config = [])
         $config += [
             'serviceDefinitionPath' => __DIR__ . '/ServiceDefinition/bigquery-v2.json',
             'componentVersion' => BigQueryClient::VERSION,
-            'apiEndpoint' => self::DEFAULT_API_ENDPOINT
+            'apiEndpoint' => null,
+            'universeDomain' => GetUniverseDomainInterface::DEFAULT_UNIVERSE_DOMAIN,
         ];
 
-        $apiEndpoint = $this->getApiEndpoint(self::DEFAULT_API_ENDPOINT, $config);
+        $apiEndpoint = $this->getApiEndpoint(null, $config, self::DEFAULT_API_ENDPOINT_TEMPLATE);
```

Comment:
Passing `null` here will break in previous versions of `google/cloud-core`, so we need to either continue to pass in  `self::DEFAULT_API_ENDPOINT`, or require these libraries to use a _minimum_ of the next version of core (in this case, `^1.53`)

Reply:
I think we should force an upgraded min `google/cloud-core` version and continue to pass the `null`.
Else if the user's client instantiation has `apiEndpoint` in the `$options`, they will need to remove it for TPC to work.

Reply:
I agree. I've bumped core to what _will_ be the new version in https://github.com/googleapis/google-cloud-php/pull/6850/commits/92b9f114369bfa0cc55431ae588ca730787ddb3e